main/src/libraries/wrdac/src/subsystems/subSystem_KARMA.cpp

```diff
@@ -169,7 +251,145 @@ bool wysiwyd::wrdac::SubSystem_KARMA::prepare()
     return true;
 }
 
-bool wysiwyd::wrdac::SubSystem_KARMA::push(const yarp::sig::Vector &targetCenter, const double theta, const double radius, const yarp::os::Bottle &options, const std::string &sName)
+bool wysiwyd::wrdac::SubSystem_KARMA::toolAttach(const std::string &armType, const Vector &dimTool)
+{
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("tool"));
+    bCmd.addVocab(yarp::os::Vocab::encode("attach"));
+    appendString(bCmd,armType);
+    appendTarget(bCmd,dimTool);
+
+    bool bReturn = sendCmd(bCmd,true);
+    std::string status;
+    bReturn ? status = "success" : status = "failed";
+
+    return bReturn;
+}
+
+void wysiwyd::wrdac::SubSystem_KARMA::toolRemove()
+{
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("tool"));
+    bCmd.addVocab(yarp::os::Vocab::encode("remove"));
+
+
+    sendCmd(bCmd,true);
+}
+
+bool wysiwyd::wrdac::SubSystem_KARMA::pushLeft(const yarp::sig::Vector &objCenter, const double &targetPosYLeft,
+                                               const double &zOff,
+                                               const std::string &armType,
+                                               const yarp::os::Bottle &options, const std::string &sName)
+{
+    // Calculate the pushing distance (radius) for push with Karma
+    double radius;
+    Vector object = objCenter;
+    Bottle opt = options;
+    selectHandCorrectTarget(opt,object);    // target is calibrated by this method
+    radius = fabs(object[1] - targetPosYLeft);
```

Comment:
Change to `double radius = fabs(object[1]-targetPosYLeft)`. It's more readable, as you immediately can see which radius is meant. Delete line 285 accordingly.